src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -34,19 +34,32 @@ public RecurringSession(SessionDate sessionDate, Duration duration, Subject subj
         this.lastSessionDate = lastSessionDate;
     }
 
-    // Checks that lastDate can be on given sessionDate.
-    private static boolean isValidEnd(SessionDate firstSessionDate, SessionDate lastSessionDate, Interval interval) {
+    /**
+     * Returns true if user input {@code firstSessionDate} and {@code interval} makes a valid {@code lastSessionDate}.
+     * @param firstSessionDate the SessionDate value for first session
+     * @param lastSessionDate the SessionDate value for last session
+     * @param interval the interval of recurrence
+     * @return true if the first session and last session is consistent, based on interval
+     */
+    public static boolean isValidEnd(SessionDate firstSessionDate, SessionDate lastSessionDate, Interval interval) {
         boolean occursAtTime = firstSessionDate.isSameTime(lastSessionDate);
         boolean occursAtDate = isConsistentDatesAndInterval(firstSessionDate, lastSessionDate, interval);
         return occursAtTime && occursAtDate;
     }
 
-    // Checks that sessionDate1 occurs on the date of sessionDate2, upon recurring 0 or more times at given interval.
-    private static boolean isConsistentDatesAndInterval(
+    /**
+     * Returns true if user input {@code firstSessionDate} and {@code interval} is consistent with Date of
```

Comment:
Hi, actually I am abit confused between this isConsistent method compared to the isValidEnd method based on the description. One mention that it returns true if it makes a valid lastSessionDate while the other returns true if it is consistent with the lastSessionDate. The 2 descriptions seems very similar in my opinion.

Reply:
yeah the only difference is if it checks the date vs both date and time

Reply:
Maybe it would be better to include this difference in the description as well. Because I think if u didn't tell me this, I would not have known from the description alone.